war/src/test/java/io/lumeer/core/facade/CollectionFacadeIntegrationTest.java

```diff
@@ -0,0 +1,269 @@
+import io.lumeer.api.model.Permission;
+import io.lumeer.api.model.Permissions;
+import io.lumeer.api.model.Project;
+import io.lumeer.api.model.Resource;
+import io.lumeer.api.model.Role;
+import io.lumeer.api.model.View;
+import io.lumeer.core.AuthenticatedUser;
+import io.lumeer.core.WorkspaceKeeper;
+import io.lumeer.core.model.SimplePermission;
+import io.lumeer.core.model.SimpleUser;
+import io.lumeer.engine.IntegrationTestBase;
+import io.lumeer.storage.api.dao.CollectionDao;
+import io.lumeer.storage.api.dao.OrganizationDao;
+import io.lumeer.storage.api.dao.ProjectDao;
+import io.lumeer.storage.api.dao.UserDao;
+import io.lumeer.storage.api.exception.ResourceNotFoundException;
+
+import org.assertj.core.api.SoftAssertions;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+@RunWith(Arquillian.class)
+public class CollectionFacadeIntegrationTest extends IntegrationTestBase {
+
+   private static final String ORGANIZATION_CODE = "TORG";
+   private static final String PROJECT_CODE = "TPROJ";
+
+   private static final String USER = AuthenticatedUser.DEFAULT_EMAIL;
+   private static final String GROUP = "testGroup";
+
+   private static final String CODE = "TCOLL";
+   private static final String NAME = "Test collection";
+   private static final String ICON = "fa-eye";
+   private static final String COLOR = "#00ee00";
+
+   private static final Permission USER_PERMISSION = new SimplePermission(USER, View.ROLES);
+   private static final Permission GROUP_PERMISSION = new SimplePermission(GROUP, Collections.singleton(Role.READ));
+
+   private static final String CODE2 = "TCOLL2";
+
+   @Inject
+   private CollectionFacade collectionFacade;
+
+   @Inject
+   private CollectionDao collectionDao;
+
+   @Inject
+   private OrganizationDao organizationDao;
+
+   @Inject
+   private ProjectDao projectDao;
+
+   @Inject
+   private UserDao userDao;
+
+   @Inject
+   private WorkspaceKeeper workspaceKeeper;
+
+   @Before
+   public void configureProject() {
+      JsonOrganization organization = new JsonOrganization();
+      organization.setCode(ORGANIZATION_CODE);
+      organization.setPermissions(new JsonPermissions());
+      Organization storedOrganization = organizationDao.createOrganization(organization);
+
+      projectDao.setOrganization(storedOrganization);
+      userDao.setOrganization(storedOrganization);
+
+      SimpleUser user = new SimpleUser(USER);
+      userDao.createUser(user);
+
+      JsonProject project = new JsonProject();
+      project.setCode(PROJECT_CODE);
+
+      JsonPermissions projectPermissions = new JsonPermissions();
+      projectPermissions.updateUserPermissions(new JsonPermission(USER, Project.ROLES.stream().map(Role::toString).collect(Collectors.toSet())));
```

Comment:
Just a hint: You can use `Role.toStringRoles(Project.ROLES)` instead

Reply:
Thanks. I have forgotten about that one.